src/FocusTrap.ts

```diff
@@ -54,13 +55,25 @@ export const FocusTrap = defineComponent({
 
   setup(props, { slots, emit }) {
     let trap: FocusTrapI | null
-    const el = ref<HTMLElement | null>(null)
+    const el = ref<HTMLElement | ComponentPublicInstance | null>(null)
 
     const ensureTrap = () => {
       if (trap) {
         return
       }
 
+      if (el === null || el?.value === null) {
+        throw new Error(
+          '[focus-trap-vue] FocusTrap requires exactly one child.'
+        )
+      }
+
+      if (!(el.value instanceof HTMLElement) && Reflect.has(el.value, '$el')) {
+        if (Reflect.get(el.value, '$el') instanceof HTMLElement) {
+          el.value = (el.value as ComponentPublicInstance).$el
```

Comment:
`el.value` should not be manually updated because it's set with a `ref` attribute.

There should probably be put in a different variable and it could avoid all the casts `as HTMLElement`

Reply:
How's this look:

```typescript
      // resolve for what the target element should be
      const targetEl: HTMLElement = (() => {
        if (!(el.value instanceof HTMLElement) && el.value?.$el) {
          if (el.value.$el instanceof HTMLElement) {
            return (el.value as ComponentPublicInstance).$el
          }
        }

        return el.value as HTMLElement
      })()

      // pass targetEl for wherever `el.value as HTMLElement` was
      const { initialFocus } = props
      trap = createFocusTrap(targetEl, { 										// <-- here
        escapeDeactivates: props.escapeDeactivates,
        allowOutsideClick: event =>
          typeof props.allowOutsideClick === 'function'
            ? props.allowOutsideClick(event)
            : props.allowOutsideClick,
        returnFocusOnDeactivate: props.returnFocusOnDeactivate,
        clickOutsideDeactivates: props.clickOutsideDeactivates,
        onActivate: () => {
          emit('update:active', true)
          emit('activate')
        },
        onDeactivate: () => {
          emit('update:active', false)
          emit('deactivate')
        },
        initialFocus: initialFocus
          ? typeof initialFocus === 'function'
            ? initialFocus()
            : initialFocus
          : targetEl,															 // <-- and here
        fallbackFocus: props.fallbackFocus,
      })
```

Reply:
let's rename `el` into wrapperElement and create a computed `el` that always return the root element and that we can pass to focus trap